src/metatensor/models/experimental/soap_bpnn/train.py

```diff
@@ -307,6 +307,7 @@ def train(
                     [average_block_by_num_atoms(targets[pa_target].block(), num_atoms)],
                 )
 
+            loss = loss_fn(predictions, targets)
```

Comment:
Can you elaborate what was going wrong here before and does this also effect the alchemical model wich basically has the same code.

Reply:
Yes, of course. That line was missing, but the variable was defined in the training loop. Essentially it was re-using the loss of the last batch in the training loop as the loss for all the batches in the validation set.
The alchemical model isn't missing that line

Reply:
I see. Is there a way to avoid this in the future? Once the next re-factoring will take place I am afraid that we reintroduce this issue.

Reply:
Yes, I've renamed the variables so that, if the line disappears again, an error will be raised

Reply:
passable solution for preventing this for now, but we need to design better tests in the future, imo